src/Stratis.Features.FederatedPeg/CollateralFederationManager.cs

```diff
@@ -16,6 +17,25 @@ public CollateralFederationManager(NodeSettings nodeSettings, Network network, I
         {
         }
 
+        public override void Initialize()
+        {
+            base.Initialize();
+
+            IEnumerable<CollateralFederationMember> collateralMembers = this.federationMembers.Cast<CollateralFederationMember>().Where(x => x.CollateralAmount != null && x.CollateralAmount > 0);
+
+            if (collateralMembers.Any(x => x.CollateralMainchainAddress == null))
+            {
+                throw new Exception("Federation can't contain members with non-zero collateral requirement but null collateral address.");
+            }
+
+            int distinctCount = collateralMembers.Select(x => x.CollateralMainchainAddress).Distinct().Count();
+
+            if (distinctCount != collateralMembers.Count())
+            {
+                throw new Exception("Federation can't contain members with duplicated collateral addresses.");
```

Comment:
Does this uniqueness condition get checked when a potential member is initially getting voted in? The vote appears to get scheduled without it being checked, is it done later?

Reply:
I've disabled voting for collateral atm. There will be a new controller for fedpeg that will accept new fed member. Old was accepting just the pubkey, now we need pubkey, address and collateral. 
So yeah, it will be done later, logic will go into new voting controller and PollResultExecutor